cubi-tools/prototypes/update_workflow.py

```diff
@@ -0,0 +1,646 @@
+#!/usr/bin/env python3
+
+import pathlib
+import sys
+import subprocess as sp
+import argparse as argp
+import os
+import shutil
+import hashlib
+import toml
+
+
+def main():
+    """
+    Main function of the 'update-workflow.py' script.
+    """
+
+    args = parse_command_line()
+    dryrun = args.dryrun
+
+    # Is it a dry run?
+    if dryrun:
+        print("\nTHIS IS A DRY RUN!!")
+
+    # check if project directory exist:
+    workflow_target = pathlib.Path(args.workflow_dir).resolve()
+    if not workflow_target.is_dir():
+        raise FileNotFoundError(
+            f"The project directory {workflow_target} doesn't exist!"
+        )
+    print(f"Project directory set as: {workflow_target}\n")
+
+    ref_repo = args.ref_repo
+    branch = args.branch
+    metadata = args.metadata
+
+    # Since using the online github repo directly to update the local workflow files
+    # is resulting in hitting an API rate limit fairly quickly a local copy is needed.
+    # The location of the template_snakemake folder holding branch/version tag
+    # of interest is on the same level as the project directory
+    workflow_branch = pathlib.Path(
+        pathlib.Path(f"{workflow_target}").resolve().parents[0],
+        "template-snakemake",
+    ).resolve()
+
+    # detect if workflow is based on CUBI's template_snakemake repo
+    # The file '/workflow/rules/commons/10_constants.smk' should be present
+    # if template_snakemake was used to create project folder
+    if pathlib.Path(
+        str(workflow_target) + "/workflow/rules/commons/10_constants.smk"
+    ).is_file():
+        pass
+    else:
+        answer_is_pos = user_response(
+            "ARE YOU SURE THIS PROJECT IS BASED ON CUBI'S "
+            "'template_snakemake' REPOSITORY"
+        )
+        if answer_is_pos:
+            pass
+        else:
+            raise NameError(
+                "This project is not based on CUBI's 'template_snakemake'. "
+                "No changes have been made!"
+            )
+
+    # Clone/update the 'template-snakemake' repo
+    # into a local folder if it exists
+    clone(workflow_target, ref_repo, branch, workflow_branch, dryrun)
+
+    # Call function to create the list of files and folders that
+    # should be made/copied/updated
+    update_files = update_file_list(workflow_branch, metadata, dryrun)[0]
```

Comment:
do not call the same function twice and access one return value at a time - call it once and assign directly to tuple

Reply:
Done